crates/core/tedge_agent/src/mqtt_operation_converter/actor.rs

```diff
@@ -0,0 +1,132 @@
+use crate::mqtt_operation_converter::error::MqttRequestConverterError;
+use async_trait::async_trait;
+use mqtt_channel::Topic;
+use tedge_actors::fan_in_message_type;
+use tedge_actors::Actor;
+use tedge_actors::DynSender;
+use tedge_actors::LoggingReceiver;
+use tedge_actors::MessageReceiver;
+use tedge_actors::RuntimeError;
+use tedge_actors::Sender;
+use tedge_api::Jsonify;
+use tedge_api::RestartOperationRequest;
+use tedge_api::RestartOperationResponse;
+use tedge_api::SoftwareListRequest;
+use tedge_api::SoftwareListResponse;
+use tedge_api::SoftwareUpdateRequest;
+use tedge_api::SoftwareUpdateResponse;
+use tedge_mqtt_ext::MqttMessage;
+
+fan_in_message_type!(AgentInput[MqttMessage, SoftwareListResponse, SoftwareUpdateResponse, RestartOperationResponse] : Debug);
+
+pub struct MqttOperationConverterActor {
+    input_receiver: LoggingReceiver<AgentInput>,
+    software_list_sender: DynSender<SoftwareListRequest>,
+    software_update_sender: DynSender<SoftwareUpdateRequest>,
+    restart_sender: DynSender<RestartOperationRequest>,
+    mqtt_publisher: DynSender<MqttMessage>,
+}
+
+#[async_trait]
+impl Actor for MqttOperationConverterActor {
```

Comment:
This could have been a `ConvertingActor` as all its transformations are simple 1:1 conversions.

Reply:
> This could have been a `ConvertingActor` as all its transformations are simple 1:1 conversions.

We could ... with one caveat. This needs a new type of sender that we have not implemented yet. One that handles several named sub-senders and dispatch messages to them depending their kind.